From 78290b1641e95304c862062ee0aca95395c5926c Mon Sep 17 00:00:00 2001
From: Josh Durgin <josh.durgin@inktank.com>
Date: Mon, 2 Jul 2012 11:55:26 -0700
Subject: [PATCH 01/24] qemu: add rbd to whitelist of migration-safe formats

QEMU (and librbd) flush the cache on the source before the
destination starts, and the destination does not read any
changeable data before that, so live migration with rbd caching
is safe.

This makes 'virsh migrate' work with rbd and caching without the
--unsafe flag.

Reported-by: Vladimir Bashkirtsev <vladimir@bashkirtsev.com>
Signed-off-by: Josh Durgin <josh.durgin@inktank.com>
---
 .mailmap                  |    3 ++-
 AUTHORS                   |    2 +-
 src/qemu/qemu_migration.c |    3 +++
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/AUTHORS b/AUTHORS
index 375db24..0e179a9 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -144,7 +144,7 @@ Patches have also been contributed by:
   Hu Tao               <hutao@cn.fujitsu.com>
   Laurent Léonard      <laurent@open-minds.org>
   MORITA Kazutaka      <morita.kazutaka@lab.ntt.co.jp>
-  Josh Durgin          <josh.durgin@dreamhost.com>
+  Josh Durgin          <josh.durgin@inktank.com>
   Roopa Prabhu         <roprabhu@cisco.com>
   Paweł Krześniak      <pawel.krzesniak@gmail.com>
   Kay Schubert         <kayegypt@web.de>
diff --git a/src/qemu/qemu_migration.c b/src/qemu/qemu_migration.c
index 48369d6..f51c99a 100644
--- a/src/qemu/qemu_migration.c
+++ b/src/qemu/qemu_migration.c
@@ -847,6 +847,9 @@ qemuMigrationIsSafe(virDomainDefPtr def)
                     continue;
                 else if (cfs < 0)
                     return false;
+            } else if (disk->type == VIR_DOMAIN_DISK_TYPE_NETWORK &&
+                       disk->protocol == VIR_DOMAIN_DISK_PROTOCOL_RBD) {
+                continue;
             }
 
             qemuReportError(VIR_ERR_MIGRATE_UNSAFE, "%s",
-- 
1.7.8.6

